src/Build/Instance/ProjectInstance.cs

```diff
@@ -326,6 +326,59 @@ public ProjectInstance(ProjectRootElement xml, IDictionary<string, string> globa
         {
         }
 
+        /// <summary>
+        /// Creates a ProjectInstance from an external created <see cref="Project"/>.
+        /// Properties and items are cloned immediately and only the instance data is stored.
+        /// </summary>
+        public ProjectInstance(Project project, ProjectInstanceSettings settings)
```

Comment:
How is this different from the other [ProjectInstance(Evaluation.Project.Data)](https://github.com/dotnet/msbuild/blob/97463e3cdb3ce4ccf9638eb68fd9f82646ca6a11/src/Build/Instance/ProjectInstance.cs#L451) constructor which is called from [Project.CreateProjectInstance](https://github.com/dotnet/msbuild/blob/97463e3cdb3ce4ccf9638eb68fd9f82646ca6a11/src/Build/Definition/Project.cs#L1333)? They seem the same to me, so I'd rather not add this additional constructor (unless I'm missing something).